File: samples/ws_study/ws_csharp/myhttppeer.cs
using System;
using SocketProAdapter;
using SocketProAdapter.ServerSide;

public class CMyHttpPeer : CHttpPeerBase
{
    protected override bool DoAuthentication(string userId, string password)
    {
        Push.Subscribe(1, 2, 7);
        Console.Write("User id = " + userId);
        Console.WriteLine(", password = " + password);
        return true; //true -- permitted; and false -- denied
    }

    protected override void OnGet()
    {
        if (Path.LastIndexOf('.') != 1)
            DownloadFile(Path.Substring(1));
        else
            SendResult("test result --- GET ---");
    }

    protected override void OnPost()
    {
        uint res = SendResult("+++ POST +++ test result");
    }

    protected override void OnUserRequest()
    {
        switch (RequestName)
        {
            case "sleep":
                int ms = int.Parse(Args[0].ToString());
                Sleep(ms);
                SendResult("");
                break;
            case "sayHello":
                SendResult(SayHello(Args[0].ToString(), Args[1].ToString()));
                break;
            case "doException":
                throw new Exception("This is test for exception");
            default:
                break;
        }
    }

    private string SayHello(string firstName, string lastName)
    {
        //notify a message to groups [2, 3] at server side
        Push.Publish("Say hello from " + firstName + " " + lastName, 2, 3);
        return "Hello " + firstName + " " + lastName;
    }

    private void Sleep(int ms)
    {
        System.Threading.Thread.Sleep(ms);
    }
}

File: tutorials/vbnet/webdemo/myhttppeer.vb
Imports System
Imports SocketProAdapter.ServerSide
Imports System.Diagnostics

Public Class CMyHttpPeer
	Inherits CHttpPeerBase

	Protected Overrides Sub OnSubscribe(ByVal groups() As UInteger)
		Console.WriteLine(UID & " subscribes for groups " & HelloWorldPeer.ToString(groups))
	End Sub

	Protected Overrides Sub OnUnsubscribe(ByVal groups() As UInteger)
		Console.WriteLine(UID & " unsubscribes for groups " & HelloWorldPeer.ToString(groups))
	End Sub

	Protected Overrides Sub OnPublish(ByVal message As Object, ByVal groups() As UInteger)
        Console.WriteLine(UID & " publishes a message (" & message.ToString() & ") to groups " & HelloWorldPeer.ToString(groups))
	End Sub

	Protected Overrides Sub OnSendUserMessage(ByVal receiver As String, ByVal message As Object)
        Console.WriteLine(UID & " sends a message (" & message.ToString() & ") to " & receiver)
	End Sub

	Protected Overrides Function DoAuthentication(ByVal userId As String, ByVal password As String) As Boolean
		Push.Subscribe(1, 2, 7)
		Console.Write("User id = " & userId)
		Console.WriteLine(", password = " & password)
		Return True 'true -- permitted; and false -- denied
	End Function

	Protected Overrides Sub OnGet()
		If Path.LastIndexOf("."c) <> 1 Then
			DownloadFile(Path.Substring(1))
		Else
			SendResult("test result --- GET ---")
		End If
	End Sub

	Protected Overrides Sub OnPost()
		Dim res As UInteger = SendResult("+++ POST +++ test result")
	End Sub

	Protected Overrides Sub OnUserRequest()
		Select Case RequestName
			Case "sleep"
				Dim ms As Integer = Integer.Parse(Args(0).ToString())
				Sleep(ms)
				SendResult("")
			Case "sayHello"
				SendResult(SayHello(Args(0).ToString(), Args(1).ToString()))
			Case Else
		End Select
	End Sub

	Private Function SayHello(ByVal firstName As String, ByVal lastName As String) As String
		'notify a message to groups [2, 3] at server side
		Push.Publish("Say hello from " & firstName & " " & lastName, 2, 3)

		Return "Hello " & firstName & " " & lastName
	End Function

	Private Sub Sleep(ByVal ms As Integer)
		System.Threading.Thread.Sleep(ms)
	End Sub
End Class